android/app/src/main/python/mmrelay/cli.py

```
@@ -0,0 +1,1614 @@
+            for field in required_fields
+            if not _is_valid_non_empty_string((credentials or {}).get(field))
+        ]
+
+        if missing_fields:
+            print(
+                f"❌ Error: credentials.json missing required fields: {', '.join(missing_fields)}"
+            )
+            print(f"   {msg_run_auth_login()}")
+            return False
+
+        return True
```

Ruff (0.12.2)

Consider moving this statement to an `else` block

(TRY300)